crml/erc20-peg/src/lib.rs

```diff
@@ -184,11 +188,20 @@ decl_module! {
 			}
 
 			let message = WithdrawMessage {
-				token_address: token_address.unwrap(),
+				token_address,
 				amount: amount.into(),
 				beneficiary
 			};
-			let event_proof_id = T::EthBridge::generate_event_proof(&message)?;
+			let event_proof_id: u64 = T::EthBridge::generate_event_proof(&message)?;
+
+			// Create a hash of withdrawAmount, tokenAddress, receiver, eventId
+			let mut buf = Vec::<u8>::with_capacity(141);
+			buf.extend_from_slice(&amount.to_le_bytes());
+			buf.extend_from_slice(&token_address.as_bytes());
+			buf.extend_from_slice(&beneficiary.as_bytes());
+			buf.extend_from_slice(&event_proof_id.to_le_bytes());
+			let withdrawal_hash: T::Hash = T::Hashing::hash(&buf);
+			ActiveWithdrawals::<T>::insert(event_proof_id, withdrawal_hash);
```

Comment:
this is the right idea.
would use the `Encode` trait so that its SCALE encoded and more easily reproducible by clients e.g.:
something like:

```suggestion
			let withdrawal_hash: T::Hash = T::Hashing::hash(&mut (message, event_poof_id).encode());
			ActiveWithdrawals::<T>::insert(event_proof_id, withdrawal_hash);
```